Aws/S3/Commands/DeleteObjectVersion.hs

```diff
@@ -0,0 +1,44 @@
+module Aws.S3.Commands.DeleteObjectVersion
+where
+
+import           Aws.Core
+import           Aws.S3.Core
+import           Data.ByteString.Char8      ({- IsString -})
+import qualified Data.Text                  as T
+import qualified Data.Text.Encoding         as T
+
+data DeleteObjectVersion = DeleteObjectVersion {
+  dovObjectName :: T.Text,
+  dovBucket :: Bucket,
+  dovVersionId :: T.Text
+}
```

Comment:
Can you please also add a "smart constructor" function, so we can more easily add new fields in the future without breaking as much existing code?